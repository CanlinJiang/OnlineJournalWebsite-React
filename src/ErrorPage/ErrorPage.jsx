import React from 'react'
import Frame from "../UtilitiesComponents/MainFrame/Frame"
import styles from "./ErrorPage.module.css"

export default function ErrorPage() {
    const BodyComponent = () => (
        <div className={styles.textContainer}>
            <h1>404</h1>
            <p>Not Found</p>
            <p>The requested URL was not found on this server</p>
        </div>
    )



    return (
        <Frame BodyComponent={BodyComponent}/>
    )
}
