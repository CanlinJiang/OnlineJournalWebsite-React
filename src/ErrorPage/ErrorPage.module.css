.textContainer {
    display: grid;
    grid-template-rows: 3;
    margin: 200px 200px 350px;
    justify-content: center;
}

.textContainer h1{
    font-size: 100px;
    margin-bottom: 20px;
}

.textContainer p{
    font-size: 17px;
}

@media (max-width:767px) {
    .textContainer {
        display: grid;
        grid-template-rows: 3;
        margin: 100px 40px 250px;
        justify-content: center;
    }
    .textContainer p{
        font-size: 16px;
    }
}

@media (min-width:1099px) {
    .textContainer h1{
        font-size: 180px;
        margin-bottom: 10px;
    }
    
    .textContainer p{
        font-size: 30px;
    }
}
