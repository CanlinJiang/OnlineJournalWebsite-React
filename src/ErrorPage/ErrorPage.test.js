import ErrorPage from "./ErrorPage"
import React from "react";

describe("ErrorPage tests", () => {
    it("renders correctly", () => {
        
      const { container } = render(<ErrorPage />);
      expect(container).toMatchSnapshot();
    });
  
  });