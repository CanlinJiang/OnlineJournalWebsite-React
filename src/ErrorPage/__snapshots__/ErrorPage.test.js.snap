// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ErrorPage tests renders correctly 1`] = `
<div>
  <div
    class="container"
  >
    <div
      class="header"
    >
      <nav
        class="navigation"
      >
        <ul
          class="leftBar"
        >
          <li>
            <a
              href="/minelist"
            >
              Mine's
            </a>
          </li>
          <li>
            <a
              href="/moodchoosing"
            >
              Moods
            </a>
          </li>
          <li>
            <a
              href="/randomwalk"
            >
              Random Walk
            </a>
          </li>
        </ul>
        <ul
          class="rightBar"
        >
          <li
            id="home"
          >
            <a
              aria-current="page"
              class="active"
              href="/"
            >
              Home
            </a>
          </li>
          <li
            id="addNewArticle"
          >
            <a
              href="/addnewarticle"
            >
              Add New Article
            </a>
          </li>
        </ul>
      </nav>
    </div>
    <div
      class="body"
    >
      <div
        class="textContainer"
      >
        <h1>
          404
        </h1>
        <p>
          Not Found
        </p>
        <p>
          The requested URL was not found on this server
        </p>
      </div>
    </div>
    <div
      class="footer"
    >
      <nav>
        <ul
          class="navigation"
        >
          <li
            id="aboutMe"
          >
            <a
              href="/aboutme"
            >
              About me
            </a>
          </li>
          <li
            id="contactMe"
          >
            <a
              href="/contactme"
            >
              Contact me
            </a>
          </li>
        </ul>
      </nav>
    </div>
  </div>
</div>
`;
